onnxruntime/core/optimizer/nhwc_transformer.cc

```diff
@@ -282,6 +283,49 @@ void NhwcTransformerImpl::TransformSplit(Node& node) {
   CreateNhwcArgument(node, node, nhwc_input->rank_);
 }
 
+void NhwcTransformerImpl::TransformPad(Node& node) {
+  auto& input_defs = node.MutableInputDefs();
+
+  auto* nhwc_input = LookupNhwcArgument(input_defs[0]);
+  if (nhwc_input == nullptr) {
+    return;
+  }
+
+  const ONNX_NAMESPACE::TensorProto* pads_tensor_proto = nullptr;
+  if (!graph_utils::NodeArgIsConstant(graph_, *input_defs[1]) ||
+      !graph_.GetInitializedTensor(input_defs[1]->Name(), pads_tensor_proto) ||
+      (pads_tensor_proto->dims_size() != 1) ||
+      (nhwc_input->rank_ != pads_tensor_proto->dims(0) / 2) ||
```

Comment:
Check if rank * 2 = proto->dims(0) instead? If dims(0) were 9, then this would still match a rank of 4, when the optimizer shouldn't touch the node at all.

Reply:
Nice. Fixed.